src/vmm/src/guest_config/common/mod.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO - Remove allow(unused) once types have been used
+
+use std::collections::{HashMap, HashSet};
+use std::ops::Add;
+
+#[cfg(target_arch = "x86_64")]
+use crate::guest_config::cpuid::Cpuid;
+
+/// Type defined to represent registers across CPU vendors
+/// (pointers and values)
+pub trait Numeric: Copy + Add<Self, Output = Self> {}
+impl Numeric for u32 {}
+impl Numeric for u64 {}
+impl Numeric for u128 {}
+
+/// Map to find and define model-specific registers
+#[cfg(target_arch = "x86_64")]
+pub type MsrRegisterMap = HashMap<RegisterPointer<u32>, RegisterValue<u32>>;
```

Comment:
"reg" is used quite a lot in our repo. Is there an advantage in using "register" instead?

Reply:
My intention here is to add clarity for anyone reading the code. Reg is very vague.